### PR TITLE
[Refactoring] Refactor BounceIntegrationTest

### PR DESCRIPTION
**Pull request by remk**
Backport refactoring done on BounceIntegrationTest in the closed #3435 PR.

### CODE REVIEW

**Comment by nvduc91** [comment]
read it

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 1]
```diff
@@ -94,21 +84,35 @@ public void dsnBounceMailetShouldDeliverBounce() throws Exception {
...
+        MailetConfiguration.Builder mailetConfiguration = MailetConfiguration.builder()
                 .matcher(All.class)
-                .mailet(Bounce.class)
-                .addProperty("passThrough", "false")))
-            .build(temporaryFolder.newFolder());
+                .mailet(mailet)
+                .addProperty("passThrough", "false");
+
+        if (additionalProperties.length > 0) {
+            ImmutableList.copyOf(additionalProperties)
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:96]: `Arrays.stream`?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java [hunk 2]
```diff
@@ -94,21 +84,35 @@ public void dsnBounceMailetShouldDeliverBounce() throws Exception {
...
+    private void setup(Class<? extends Mailet> mailet, Pair<String, String>... additionalProperties) throws Exception {
+        MailetConfiguration.Builder mailetConfiguration = MailetConfiguration.builder()
                 .matcher(All.class)
-                .mailet(Bounce.class)
-                .addProperty("passThrough", "false")))
-            .build(temporaryFolder.newFolder());
+                .mailet(mailet)
+                .addProperty("passThrough", "false");
+
+        if (additionalProperties.length > 0) {
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/BounceIntegrationTest.java:95]: why do you need that?

**Comment by Arsnael** [comment]
```
[5c263148ee1c925643095188bde5fff5fe546165] [ERROR] Errors: 
[5c263148ee1c925643095188bde5fff5fe546165] [ERROR]   MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout
```

**Comment by Arsnael** [comment]
test this please

**Comment by Arsnael** [comment]
Log output lost

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**